### PR TITLE
Impliment new header and footer design

### PR DESCRIPTION
**Pull request by gemfarmer**
For #544 based on @andrewmaier's [prototype header and footer](https://pages.18f.gov/micropurchase/)

What was done: 

Header
---
- reduced logo
- adjusted font sizes
- slightly refactored html and css
- made logo clickable
- adjusted background color

Footer
---
- restructure of html so that main links are in the main area of the footer
- adjusted font sizes and colors
- reduced logo size (same as in header)
- adjusted background color

Misc
---
- broke out header, footer, disclaimer into includes (_finally_)
- adjusted spacing on below main content, additional spacing in auction details view
- updated site-wide link highlight color
- added `.scss-lint.yml` file

Todo still:
--- 
- redo login button in accordance with Andrew's design
- make login button work with dropdown
- restyle drawer to align with new header/footer

### CODE REVIEW

**Comment by gemfarmer** [comment]
@mtorres253 @andrewmaier could I get a review?

Update: The header and footer are partially up to date with Andrew's prototype. I did not add the elegant login button in the upper right hand corner, but since we have yet to decide what makes sense there, I left it untouched for now.

I think this could be merged (pending review) as a preliminary change, if only for the much-improved footer. 

Note: There are 4 outstanding warnings on codeclimate. I am comfortable with all of these issues.

Header: 
---
screen shot 2016-05-16 at 4 43 36 pm
screen shot 2016-05-16 at 4 43 23 pm

Footer:
---

screen shot 2016-05-16 at 4 44 01 pm
screen shot 2016-05-16 at 4 43 45 pm

File: .scss-lint.yml [hunk 1]
```diff
@@ -0,0 +1,253 @@
+# Default application configuration that all configurations inherit from.
+
+scss_files: "**/*.scss"
```
Comment (jessieay) [inline .scss-lint.yml]: yesssss linters. love it

File: app/views/components/_user_nav_drawer.html.erb [hunk 2]
```diff
@@ -14,3 +14,12 @@
       </li>
   </ul>
 </nav>
+
+<!-- <a class="nav-sign-out" href="/micropurchase/sign-in/">
```
Comment (jessieay) [inline app/views/components/_user_nav_drawer.html.erb]: remove?

File: features/step_definitions/feedback_steps.rb [hunk 3]
```diff
@@ -2,8 +2,8 @@
   expect(page).to have_link('Feedback')
 end
 
-Then(/^I should see a link to get in touch$/) do
-  expect(page).to have_content('Get In Touch')
+Then(/^I should see an email link to get in touch$/) do
+  expect(page).to have_content('micropurchase@gsa.gov')
```
Comment (jessieay) [inline features/step_definitions/feedback_steps.rb]: is this a mailto link? http://stackoverflow.com/a/10753324/1019369

File: app/views/components/_user_nav_drawer.html.erb (hunk 2 above)
Comment (gemfarmer) [inline app/views/components/_user_nav_drawer.html.erb]: Normally I would say yes, but this commented out code is going to be added soon, so it is essentially staged here.

File: features/step_definitions/feedback_steps.rb (hunk 3 above)
Comment (gemfarmer) [inline features/step_definitions/feedback_steps.rb]: @jessieay that stackoverflow implementation seems like overkill for this instance, but all for using the `has_xlink` method. more specific.

**Comment by gemfarmer** [comment]
@jessieay @mtorres253 AFIAK this is ready to merge. Could one of you take a look and merge if you agree?

**Comment by gemfarmer** [comment]
@jessieay of the 4 lint errors, 1 is catching inaccurately. It says I have `0.5` written, but it is actually `.5`
The other 3 are lints that I want to have, but in this case I don't think they make sense.

File: app/assets/stylesheets/blocks/_header.scss [hunk 4]
```diff
@@ -45,21 +58,25 @@ nav.usa-site-navbar a {
 }
 
 .header-title-text {
-  font-size: 1.5rem;
+  font-size: 2rem;
+  font-weight: $font-light;
+  letter-spacing: -0.5px;
```
Comment (jessieay) [inline app/assets/stylesheets/blocks/_header.scss]: code climate wants you to write this without the 0

File: app/assets/stylesheets/components/_nav-button.scss [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  line-height: 1.4rem;
+  padding: 0;
+  position: relative;
+  text-align: left;
+  text-decoration: none;
+
+  &:hover,
+  &:focus {
+    background: $login-btn-bg-primary;
+    background: linear-gradient($login-btn-bg-primary, $login-btn-bg-secondary);
```
Comment (jessieay) [inline app/assets/stylesheets/components/_nav-button.scss]: confused about why we have this twice for one selector?

**Comment by gemfarmer** [comment]
The background actually _needs_ to be there because it is a browser compatibility fail-safe, but I generally want scss-lint to catch when I'm doubling up on styles :shrug:

File: app/assets/stylesheets/components/_nav-button.scss (hunk 5 above)
Comment (gemfarmer) [inline app/assets/stylesheets/components/_nav-button.scss]: linear-gradient isn't always compatible with browsers, so having a back-up that works is important

File: app/views/components/_nav_drawer.erb [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<drawer class="usa-width-one-fourth drawer" aria-hidden="true">
```
Comment (jessieay) [inline app/views/components/_nav_drawer.erb]: I believe this file is unused (removed in https://github.com/18F/micropurchase/pull/583)

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -47,37 +47,11 @@
...
-          <button id="button-nav-drawer" class="usa-button usa-button-outline-inverse" data-drawer-toggle="menu"><span class="fa fa-bars"></span></button>
-          <div class="header-account">
-            <%= render partial: current_user.decorate.nav_drawer_partial,
-              locals: { current_user: current_user } %>
-          </div>
-        </div>
-      </div>
-    </nav>
-  </header>
+  <%= render partial: 'components/nav_drawer', locals: { current_user: current_user } %>
```
Comment (jessieay) [inline app/views/layouts/application.html.erb]: this can be removed -- isn't rendering on page

File: app/views/components/_nav_drawer.erb (hunk 6 above)
Comment (gemfarmer) [inline app/views/components/_nav_drawer.erb]: It is used. It is the mobile navigation. Removing it in #583 broke the drawer.

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -86,53 +60,8 @@
...
-              <a class="usa-footer-primary-link" href="<%= faq_path %>">Policies</a>
-            </li>
-            <li class="usa-footer-primary-content">
-              <a class="usa-footer-primary-link" href="https://pages.18f.gov/micropurchase-api-docs/">API Docs</a>
-            </li>
-          </ul>
-        </nav>
-      </div>
-    </div>
+  <%= render partial: "components/site_footer" %>
```
Comment (jessieay) [inline app/views/layouts/application.html.erb]: yesss so glad we are getting these into well-named partials

File: features/step_definitions/feedback_steps.rb [hunk 9]
```diff
@@ -2,8 +2,10 @@
   expect(page).to have_link('Feedback')
 end
 
-Then(/^I should see a link to get in touch$/) do
-  expect(page).to have_content('Get In Touch')
+Then(/^I should see an email link to get in touch$/) do
+  mailto_link = '//a[@href="mailto:micropurchase@gsa.gov"]'
+  expect(page).to have_xpath(mailto_link)
+  expect(page).to have_content('micropurchase@gsa.gov')
```
Comment (jessieay) [inline features/step_definitions/feedback_steps.rb]: 😄

**Comment by jessieay** [comment]
@gemfarmer two comments about removing unused partial and ref to that partial, otherwise we are GOOD TO GO

File: app/views/components/_nav_drawer.erb (hunk 6 above)
Comment (gemfarmer) [inline app/views/components/_nav_drawer.erb]: https://github.com/18F/micropurchase/blob/bh-header-footer/app/views/layouts/application.html.erb#L51

File: app/views/layouts/application.html.erb (hunk 7 above)
Comment (gemfarmer) [inline app/views/layouts/application.html.erb]: @jessieay it is rendering, it just isn't viewable unless you are in mobile and click on the menu bars icon

File: app/views/layouts/application.html.erb (hunk 8 above)
Comment (gemfarmer) [inline app/views/layouts/application.html.erb]: yeahhh. Not sure why I didn't do that originally. laziness I guess?

**Comment by gemfarmer** [comment]
@jessieay just left you comment on those! They are here to stay (at least until we rethink that header region). Shall I merge?

**Pull request merged**